### PR TITLE
Feature/gap 1913 feature flags for one login

### PR DESCRIPTION
**Pull request by ryan-tco**
## Description

Ticket # GAP-1913

Summary of the changes and the related issue. List any dependencies that are required for this change:

Adds feature flag logic around login and logout steps of user journey to prepare for OneLogin toggle 

## Type of change

Please check the relevant options.

- [x] New feature (non-breaking change which adds functionality).

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [x] Unit Test

# Checklist:

- [x] If I have listed depedenencies above, I have ensured that they are present in the target branch.
- [x] I have performed a self-review of my code.
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

### CODE REVIEW

File: packages/admin/src/middleware.test.ts [hunk 1]
```diff
@@ -6,11 +6,16 @@ import '@testing-library/jest-dom';
 import { middleware } from './middleware.page';
 import { NextRequest, NextResponse } from 'next/server';
 
+jest.mock('./utils/general.ts', () => ({
+  getLoginUrl: () => 'http://localhost:8082/login',
+}));
```
Comment (jgunnCO) [inline packages/admin/src/middleware.test.ts:11]: Do you need to mock this if you're feeding in the same URL via env?

File: packages/admin/src/middleware.page.ts [hunk 2]
```diff
@@ -42,6 +43,6 @@ export function middleware(req: NextRequest, ev: NextFetchEvent) {
 
     return res;
   } else {
-    return NextResponse.redirect(process.env.LOGIN_URL!);
+    return NextResponse.redirect(getLoginUrl()!);
```
Comment (jgunnCO) [inline packages/admin/src/middleware.page.ts:46]: if we move the non-null assertions inside `getLoginUrl` we don't need to add them everywhere it's called like this.

Comment (jgunnCO) [reply packages/admin/src/middleware.page.ts:46]:
```js
const getLoginUrl = () => 
  process.env.ONE_LOGIN_ENABLED ? process.env.V2_LOGIN_URL! : process.env.LOGIN_URL!;

```

File: packages/applicant/src/utils/general.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const getLoginUrl = () => {
+    return process.env.ONE_LOGIN_ENABLED ? process.env.V2_LOGIN_URL : process.env.LOGIN_URL;
+  };
+
+    export { getLoginUrl };
```
Comment (jgunnCO) [inline packages/applicant/src/utils/general.ts:5]: Formatting in this file needs fixed up. I've got a PR up to add prettier into our lint config, I'm just fixing that up so I can merge it now.

File: packages/admin/.env.example [hunk 4]
```diff
@@ -9,4 +9,7 @@ BACKEND_HOST=http://localhost:8080
 COOKIE_SECRET=E6AAAB815903D8CDD81246EFC8275C13EE34541134092A635572C8F7F47448BE
 SPOTLIGHT_URL=https://cabinetoffice-spotlight.force.com/s/login/
 TINYMCE_API_KEY=
-FEATURE_ADVERT_BUILDER=enabled
\ No newline at end of file
+FEATURE_ADVERT_BUILDER=enabled
+ONE_LOGIN_ENABLED=true
```
Comment (john-tco) [inline packages/admin/.env.example:13]: if we set this to `false` instead it is going to be truthy when we check it inside the `isLoggedIn`function (it's a string)

**Comment by john-tco** [comment]
Do we need to add the flag in the Applicant app too?

**Comment by ryan-tco** [comment]
Added the env vars to the infra pipeline

File: packages/admin/src/middleware.page.ts (hunk 2 above)
Comment (ryan-tco) [reply packages/admin/src/middleware.page.ts]: > We can avoid non-null assertions here too if we provide a definition `environment.d.ts`.
> 
> ```
> declare global {
>   namespace NodeJS {
>     interface ProcessEnv {
>       //...
>       ONE_LOGIN_ENABLED: string
>     }
>   }
> }
> 
> export { }
> ```

Would this be better solved in a ticket to remove all non-null assertions?

File: packages/applicant/src/utils/general.ts (hunk 3 above)
Comment (ryan-tco) [reply packages/applicant/src/utils/general.ts:5]: Happy to merge develop into the branch and run prettier when your pr is in

Comment (jgunnCO) [reply packages/applicant/src/utils/general.ts:5]: PR merged now.

File: packages/admin/src/middleware.page.ts (hunk 2 above)
Comment (jgunnCO) [reply packages/admin/src/middleware.page.ts]: I think non-null assertions in general are fine, but we can avoid the need here specifically.

Comment (john-tco) [reply packages/admin/src/middleware.page.ts]: have added those types

**Pull request merged**